src/prefect/records/base.py

```diff
@@ -22,6 +22,16 @@ class TransactionRecord:
 
 
 class RecordStore(abc.ABC):
+    @property
+    def persists_result_data(self) -> bool:
+        """
+        Check if the record store persists result data.
+
+        Returns:
+            bool: True if the record store persists result data; False otherwise.
+        """
+        return False
```

Comment:
This allows a transaction to communicate to an actor interested in persisting results if they should persist or delegate persistence to the transaction. If `False`, the interested actor should persist the result without metadata as the transaction is always assumed to handle result metadata persistence.

This feels janky and hard to understand, so I'm very open to alternative approaches for communicating this information.